Makefile

```
@@ -1,51 +1,77 @@
-OS_NAME := $(shell uname -s | tr A-Z a-z)
-EXEC = bin/LRez
-ifeq ($(OS_NAME),linux)
-        SHLIB_EXT=so
-else
-        SHLIB_EXT=dylib
+ifeq ($(SHLIB_EXT),)
+	OS_NAME := $(shell uname -s | tr A-Z a-z)
+	ifeq ($(OS_NAME),linux)
+		SHLIB_EXT=.so
+	else
+		SHLIB_EXT=.dylib
+	endif
 endif
 
 curDir = $(shell pwd)
-CC = g++
-CFLAGS  = -Wall -pedantic -O3 -m64 -shared -Wl,--whole-archive -lpthread -Wl,--no-whole-archive -std=c++11 -fPIC
 
-BAMTOOLS_INC = $(curDir)/bamtools/include/bamtools/
-BAMTOOLS_LIB = $(curDir)/bamtools/lib/
+PREFIX ?= /usr/local
+BINDIR = $(PREFIX)/bin
+LIBDIR = $(PREFIX)/lib
+
+CXX ?= g++
+CXXFLAGS += -Wall -pedantic -O3 -m64 -std=c++11 -fPIC
+LDFLAGS += -L./lib -Wl,-rpath,$(LIBDIR) -Wl,--whole-archive -lpthread -Wl,--no-whole-archive
+
+BAMTOOLS_LIB_PREFIX = lrez_
+BAMTOOLS_LIB = lib/lib$(BAMTOOLS_LIB_PREFIX)bamtools$(SHLIB_EXT)
 
-LREZ_INC = $(curDir)/src/include/
-LREZ_LIB = $(curDir)/lib/
+BAMTOOLS_INC = ./include/bamtools/
+LREZ_INC = ./src/include/
 
-LDFLAGS_BAMTOOLS = -lbamtools -L$(BAMTOOLS_LIB)
-LDFLAGS_LREZ = -llrez -L$(LREZ_LIB)
-LDFLAGS_BOOST_LZ_LM = -lboost_iostreams -lz -lm -lc
+LIBS_LREZ = -llrez
+LIBS_BOOST_LZ_LM = -lboost_iostreams -lz -lm -lc
+LIBS_BAMTOOLS = -l$(BAMTOOLS_LIB_PREFIX)bamtools
 
 MAIN = src/main.o
 REVCOMP = src/reverseComplement.o
 SOURCE = src/alignmentsRetrieval.o src/barcodesComparison.o src/barcodesExtraction.o src/indexManagementBam.o src/utils.o src/indexManagementFastq.o src/readsRetrieval.o src/gzIndex.o
 SUBCOMMANDS = src/subcommands/compare.o src/subcommands/extract.o src/subcommands/help.o src/subcommands/indexBam.o src/subcommands/queryBam.o src/subcommands/indexFastq.o src/subcommands/queryFastq.o
 
 EXEC = bin/LRez
-LIB = lib/liblrez.${SHLIB_EXT}
-
-all: directories $(LIB) $(EXEC)
+LIB = lib/liblrez${SHLIB_EXT}
 
+all: $(LIB) $(EXEC)
 
 directories:
 	mkdir -p bin/ lib/
 
-$(LIB): $(SUBCOMMANDS) $(SOURCE) $(REVCOMP)
-	$(CC) -fPIC -shared -o $(LIB) $(SUBCOMMANDS) $(SOURCE) $(REVCOMP) $(LDFLAGS_BAMTOOLS) -Wl,-rpath,$(BAMTOOLS_LIB) $(LDFLAGS_BOOST_LZ_LM)
+$(BAMTOOLS_LIB):
+	mkdir -p bamtools/build
+	cd bamtools/build && \
+		cmake \
+			-DBUILD_SHARED_LIBS=ON \
+			-DCMAKE_SHARED_LIBRARY_PREFIX_CXX=lib$(BAMTOOLS_LIB_PREFIX) \
+			-DCMAKE_INSTALL_LIBDIR=lib \
+			-DCMAKE_INSTALL_PREFIX=$(curDir) \
+			.. && \
+		$(MAKE) -C src/api && \
+		cmake -DCMAKE_INSTALL_LOCAL_ONLY=1 -P src/cmake_install.cmake && \
+		cmake -P src/api/cmake_install.cmake
```

Okay, I went a little overboard here :).
`$(MAKE) && $(MAKE) install` would probably also work and be less cryptic. I was just interested if one could tell CMake to just build/install the library and headers :X. Probably it makes sense to simplify this, because it's not much extra work that is happening there.

---

I just tried it out and
```suggestion
		$(MAKE) && \
		$(MAKE) install
```
works perfectly fine too and should probably be used instead because simpler and less brittle ;).